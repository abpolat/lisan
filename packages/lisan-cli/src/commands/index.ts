import * as yargs from 'yargs';
import * as compile from './compile';

const commands: { [key: string]: yargs.CommandModule } = {
  compile,
};

const registerCommands = (): void =>
  Object.keys(commands).forEach((key: string): void => {
    yargs.command(commands[key]);
  });

export { registerCommands };
