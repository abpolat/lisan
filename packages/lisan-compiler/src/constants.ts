const FORBIDDEN_VARIABLE_NAMES = Object.freeze(['t', 'c']);
const FORBIDDEN_FUNCTION_NAMES = Object.freeze(['eval']);

const LOCALIZATION_METHODS = Object.freeze([
  'currency',
  'ordinal',
  'number',
  'dateTime',
  'dateShort',
  'dateMedium',
  'dateLong',
  'dateFull',
  'timeShort',
  'timeMedium',
  'timeLong',
]);

const ERRORS = Object.freeze({
  InvalidTemplateLiteral:
    'Provided Lisan literal has to be a valid javascript template literal.',
  InvalidMemberExpression:
    'A member expression cannot contain call expressions or dynamic property accessing!',
  InvalidExpression: 'Entry contains invalid Expression statement.',
  // @todo improve this error message with list of allowed functions
  invalidCallExpression: 'Entry contains invalid call expression.',
  InvalidFormatterFnArgumentCount:
    'Formatters cannot take more than 1 arguments.',
  InvalidTFnArgumentCount: '"t" function cannot take more than 2 arguments.',
  InvalidCFnArgumentCount: '"c" function cannot take more than 3 arguments.',
  InvalidFnArgument:
    'Provided argument can only be an object expression with shorthand properties. Example: `{prop1, prop2, prop3}`',
  EntryKeyMustBeString: 'Entry key must be valid string.',
  FirstArgumentMustBeAValidIdentifier:
    'First argument must be valid identifier. Example: `timeShort(myValue)`',
  SecondArgumentMustBeObjectExpression:
    'Second argument must be an object expression.',
  ThirdArgumentMustBeObjectExpression:
    'Third argument must be an object expression.',
  ConditionalValueIsMandatory:
    '"c" requires a conditional value pick conditional entry.',
  ForbiddenVariableName: (name: string) =>
    `Lisan literal contains a forbidden variable name: "${name}".`,
  ForbiddenFunctionName: (name: string) =>
    `Lisan literal contains a forbidden variable name: "${name}".`,
  ConflictingFunctionName: (name: string) =>
    `Lisan literal contains a function with same variable name: "${name}".`,
  BacktickNotAllowed:
    'Backticks are not allowed in lisan literals. Please use backspace "\\" to escape backtick characters. Example: "\\`"',
  NoParentKeyCall:
    'Lisan literal cannot have a function call with its own entry key',
});

export {
  ERRORS,
  FORBIDDEN_VARIABLE_NAMES,
  LOCALIZATION_METHODS,
  FORBIDDEN_FUNCTION_NAMES,
};
