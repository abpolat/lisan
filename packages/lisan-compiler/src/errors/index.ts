import ValidationError from './validation-error';
import ParseError from './parse-error';

export { ValidationError, ParseError };
