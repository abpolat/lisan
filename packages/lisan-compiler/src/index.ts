import { parseLisanLiteral } from './lib/parse-lisan-literal';
import { parse } from './lib/parse';
import { generate } from './lib/generate';

export { parseLisanLiteral, parse, generate };
