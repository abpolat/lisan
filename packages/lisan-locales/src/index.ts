export { default as enUS } from './en-us';
export { default as tr } from './tr';
