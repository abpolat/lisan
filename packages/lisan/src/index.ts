import Lisan from './Lisan';

const lisan = new Lisan();
const t = lisan.t.bind(lisan);
const c = lisan.c.bind(lisan);

export { lisan, t, c, Lisan };
