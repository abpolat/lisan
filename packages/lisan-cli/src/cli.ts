import * as yargs from 'yargs';
import * as findUp from 'find-up';
import * as path from 'path';
import { getConfigFromFile } from './config';
import { registerCommands } from './commands';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const pkg = require('../package.json');

registerCommands();

const configPath = findUp.sync(['.lisanrc', '.lisanrc.json']) || '';

// init program
yargs
  .scriptName('lisan')
  .usage('Usage: $0 <command> [options]')
  .epilogue('For more information, please visit http://lisanjs.com')
  .alias('v', 'version')
  .config(getConfigFromFile(configPath))
  .option('c', {
    alias: 'config',
    desc: 'Path to JSON config file',
    default: path.relative(process.cwd(), configPath),
  })
  .version(pkg.version)
  .alias('h', 'help')
  .help('help')
  .showHelpOnFail(false, 'Specify --help for available options');

export default yargs;
