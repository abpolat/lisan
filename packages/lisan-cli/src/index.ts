/* istanbul ignore file */
import cli from './cli';

const { argv } = cli;

if (!argv._[0]) {
  cli.showHelp();
}
