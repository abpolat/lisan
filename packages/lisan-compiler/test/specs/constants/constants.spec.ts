import * as constants from '../../../src/constants';

describe('Constants', () => {
  it('should always match the snapshot!', () => {
    expect(constants).toMatchSnapshot();
  });
});
