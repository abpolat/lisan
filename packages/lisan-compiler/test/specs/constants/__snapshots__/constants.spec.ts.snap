// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Constants should always match the snapshot! 1`] = `
Object {
  "ERRORS": Object {
    "BacktickNotAllowed": "Backticks are not allowed in lisan literals. Please use backspace \\"\\\\\\" to escape backtick characters. Example: \\"\\\\\`\\"",
    "ConditionalValueIsMandatory": "\\"c\\" requires a conditional value pick conditional entry.",
    "ConflictingFunctionName": [Function],
    "EntryKeyMustBeString": "Entry key must be valid string.",
    "FirstArgumentMustBeAValidIdentifier": "First argument must be valid identifier. Example: \`timeShort(myValue)\`",
    "ForbiddenFunctionName": [Function],
    "ForbiddenVariableName": [Function],
    "InvalidCFnArgumentCount": "\\"c\\" function cannot take more than 3 arguments.",
    "InvalidExpression": "Entry contains invalid Expression statement.",
    "InvalidFnArgument": "Provided argument can only be an object expression with shorthand properties. Example: \`{prop1, prop2, prop3}\`",
    "InvalidFormatterFnArgumentCount": "Formatters cannot take more than 1 arguments.",
    "InvalidMemberExpression": "A member expression cannot contain call expressions or dynamic property accessing!",
    "InvalidTFnArgumentCount": "\\"t\\" function cannot take more than 2 arguments.",
    "InvalidTemplateLiteral": "Provided Lisan literal has to be a valid javascript template literal.",
    "NoParentKeyCall": "Lisan literal cannot have a function call with its own entry key",
    "SecondArgumentMustBeObjectExpression": "Second argument must be an object expression.",
    "ThirdArgumentMustBeObjectExpression": "Third argument must be an object expression.",
    "invalidCallExpression": "Entry contains invalid call expression.",
  },
  "FORBIDDEN_FUNCTION_NAMES": Array [
    "eval",
  ],
  "FORBIDDEN_VARIABLE_NAMES": Array [
    "t",
    "c",
  ],
  "LOCALIZATION_METHODS": Array [
    "currency",
    "ordinal",
    "number",
    "dateTime",
    "dateShort",
    "dateMedium",
    "dateLong",
    "dateFull",
    "timeShort",
    "timeMedium",
    "timeLong",
  ],
}
`;
